retroshare-gui/src/gui/MainWindow.cpp

```diff
@@ -673,31 +687,49 @@ void MainWindow::createMenuBar()
     actionMinimize->setShortcutContext(Qt::ApplicationShortcut);
     actionMinimize->setShortcut(QKeySequence(Qt::CTRL + Qt::Key_M));
     actionMinimize->setShortcutVisibleInContextMenu(true);
-    connect(actionMinimize,SIGNAL(triggered()),this,SLOT(showMinimized())) ;
+    connect(actionMinimize,SIGNAL(triggered()),this,SLOT(minimizeWindow())) ; 
+
+    actionCloseWindow = new QAction(tr("Close window"),this);
+    actionCloseWindow->setShortcutContext(Qt::ApplicationShortcut);
+    actionCloseWindow->setShortcut(QKeySequence::Close);
+    actionCloseWindow->setShortcutVisibleInContextMenu(true);
+    connect(actionCloseWindow,SIGNAL(triggered()),this,SLOT(closeWindow())) ;
 
     menuBar = new QMenuBar(this);
     QMenu *fileMenu = menuBar->addMenu("");
     fileMenu->addAction(actionMinimize);
+    fileMenu->addAction(actionCloseWindow);
 
     dockMenu = new QMenu(this);
     dockMenu->setAsDockMenu();
-    dockMenu->addAction(tr("Open Messages"), this, SLOT(Mess()));
+    dockMenu->addAction(tr("Open Messages"), this, SLOT(showMess()));
     dockMenu->addAction(tr("Bandwidth Graph"), this, SLOT(showBandwidthGraph()));
     dockMenu->addAction(tr("Statistics"), this, SLOT(showStatisticsWindow()));
     dockMenu->addAction(tr("Options"), this, SLOT(showSettings()));
     dockMenu->addAction(tr("Help"), this, SLOT(showHelpDialog()));
 
-    dockMenu->addSeparator();
-    QObject::connect(dockMenu, SIGNAL(aboutToShow()), this, SLOT(updateMenu()));
-    toggleVisibilityAction = dockMenu->addAction(tr("Show/Hide"), this, SLOT(toggleVisibilitycontextmenu()));
```

Comment:
so the "show/hide" function is not there anymore? Hoe do you show the window when it's closed now?

Reply:
i added this only temporary removed it now , macos has his own show/hide and now the click from dock now works and docks own show/hide works, we doesnt need that more that always in systemtray menu own show/hide

Reply:
![image](https://github.com/RetroShare/RetroShare/assets/9952056/0d2a4fda-4a69-4667-95d2-3045180a1b32)
since i added the fix with dock icon handler rs comes from Hide/show or from one click